aries/aries_vcx_core/src/wallet2/mod.rs

```diff
@@ -0,0 +1,77 @@
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+
+use crate::errors::error::VcxCoreResult;
+
+pub enum SigType {
+    EdDSA,
+    ES256,
+    ES256K,
+    ES384,
+}
+
+impl From<SigType> for &str {
+    fn from(value: SigType) -> Self {
+        match value {
+            SigType::EdDSA => "eddsa",
+            SigType::ES256 => "es256",
+            SigType::ES256K => "es256k",
+            SigType::ES384 => "es384",
+        }
+    }
+}
+
+#[async_trait]
+pub trait Wallet: RecordWallet + DidWallet {}
+
+#[async_trait]
+pub trait DidWallet {
+    type DidAttrs;
+    type CreatedDid;
+    type DidKey;
+    type KeyAttrs;
+
+    async fn create_key(&self, key_attrs: Self::KeyAttrs) -> VcxCoreResult<()>;
+
+    async fn create_did(&self, attrs: Self::DidAttrs) -> VcxCoreResult<Self::CreatedDid>;
+
+    async fn did_key(&self, did: &str) -> VcxCoreResult<Self::DidKey>;
+
+    async fn replace_did_key(&self, did: &str) -> VcxCoreResult<Self::DidKey>;
+
+    async fn sign(
+        &self,
+        verkey_name: &str,
+        msg: &[u8],
+        sig_type: SigType,
+    ) -> VcxCoreResult<Vec<u8>>;
+
+    async fn verify(
+        &self,
+        vk: &str,
+        msg: &[u8],
+        signature: &[u8],
+        sig_type: SigType,
+    ) -> VcxCoreResult<bool>;
+}
+
+#[async_trait]
+pub trait RecordWallet {
+    type Record;
+    type RecordId;
+    type FoundRecord;
```

Comment:
It would be rare, but they might be. Askar allows specifying expiry when creating/updating a record, but expiry is not returned when fetching the record.